frontend/src/pages/AccountManagement/index.js

```diff
@@ -41,6 +41,28 @@ const frequencyValues = [
   { key: 'this month', label: 'Monthly digest' },
 ];
 
+const submitsForApprovalRoles = [
+  'ECM',
+  'GSM',
+  'TTAC',
+];
+
+const managerAndCollaboratorRoles = [
+  'ECM',
+  'ECS',
+  'FES',
+  'GS',
+  'GSM',
+  'HS',
+  'TTAC',
+];
+
+const recipientsAvailable = [
+  'PS',
+  'SPS',
+  'GMS',
+];
+
 const emailTypesMap = [
```

Comment:
Can I suggest something like this? I don't think emailTypesMap[0] emailTypesMap[1] etc reads very well

```js
const emailTypesMap = {
  submitsForApprovalRoles: [{
    name: '',
    description: 'Someone submits an activity report for my approval.',
    keyName: 'emailWhenReportSubmittedForReview',
  }],
  managerAndCollaboratorRoles: [{
    name: '',
    description: 'A manager requests changes to an activity report that I created or collaborated on.',
    keyName: 'emailWhenChangeRequested',
  },
  {
    name: '',
    description: 'Managers approve an activity report that I created or collaborated on.',
    keyName: 'emailWhenReportApproval',
  },
  {
    name: '',
    description: 'I\'m added as a collaborator on an activity report.',
    keyName: 'emailWhenAppointedCollaborator',
  }],
  recipientsAvailable: [{
    name: 'Program Specialists only',
    description: 'One of my recipients\' activity reports is available.',
    keyName: 'emailWhenRecipientReportApprovedProgramSpecialist',
  }],
};

const getEmailOptionsByUserRoles = (roles) => {
  const userRoles = roles.map((role) => role.name);
  const userEmailOptions = [];

  if (userRoles.length) {
  // If role names contains any of the roles in allEmailRoles, add ar for approval.
    if (userRoles.some((role) => submitsForApprovalRoles.includes(role))) {
      userEmailOptions.concat(emailTypesMap.submitsForApprovalRoles);
    }

    // If role names contains any of the roles in allEmailRoles, add ar for change request.
    if (userRoles.some((role) => managerAndCollaboratorRoles.includes(role))) {
      userEmailOptions.concat(emailTypesMap.managerAndCollaboratorRoles);
    }

    if (userRoles.some((role) => recipientsAvailable.includes(role))) {
      userEmailOptions.concat(emailTypesMap.recipientsAvailable);
    }
  }

  // return returnEmailOptions;
  return userEmailOptions;
};

Reply:
updated